exporter.go

```diff
@@ -639,17 +641,67 @@ samples:
 }
 
 type StatsDUDPListener struct {
-	conn *net.UDPConn
+	conn          *net.UDPConn
+	bufferWatcher *telemetry.BufferWatcher
+	bufferWarning uint32
 }
 
-func (l *StatsDUDPListener) Listen(e chan<- Events) {
+func (l *StatsDUDPListener) Listen(threadCount int, packetHandlers int, e chan<- Events) {
+	bufferWatcher, err := telemetry.NewBufferWatcher(l.conn)
+	l.bufferWatcher = bufferWatcher
+	shouldWatchBuffer := true
+
+	if err != nil {
+		shouldWatchBuffer = false
+		log.Debugf("Unable to watch UDP buffer size due to: %v", err)
```

Comment:
would this always happen on non-Linux platforms?

Reply:
Would this always log? Yes, it would. I figured a debug log would be a good place to quietly drop that to minimize confusion over why they can't get the buffer tracking to work, but if you think it's unnecessary, I would be happy to more quietly short circuit that.

Reply:
it's fine as it is, just trying to understand! Maybe add a comment to say how we expect the initialization to behave on non-Linux systems?